src/widgets/SettingsIcon.js

```diff
@@ -0,0 +1,153 @@
+
+      <PaperModalContainer isVisible={settingsModalOpen} onClose={toggleSettingsModal}>
+        <View style={localStyles.container}>
+          <FlexRow
+            alignItems="flex-start"
+            justifyContent="flex-end"
+            style={localStyles.cancelContainer}
+            flex={0}
+          >
+            <IconButton onPress={toggleSettingsModal} Icon={<CancelIcon />} />
+          </FlexRow>
+
+          <Text style={localStyles.heading}>{modalStrings.permissions}</Text>
+          <PermissionRow
+            onPress={requestWriteStorage}
+            label={modalStrings.storage_permission}
+            comment={modalStrings.storage_permission_comment}
+            enabled={writeStorage}
+          />
+          <PermissionRow
+            onPress={requestLocation}
+            label={modalStrings.location_permission}
+            comment={modalStrings.location_permission_comment}
+            enabled={location}
+          />
+          <Text style={localStyles.heading}>{modalStrings.services}</Text>
+          <PermissionRow
+            onPress={requestBluetooth}
+            label={modalStrings.bluetooth_permission}
+            comment={modalStrings.bluetooth_permission_comment}
+            enabled={bluetooth}
+          />
+        </View>
+      </PaperModalContainer>
+    </>
+  );
+};
+
+const localStyles = StyleSheet.create({
+  cancelContainer: { padding: 5, width: '100%' },
+  container: {
+    justifyContent: 'flex-start',
+    flex: 1,
+    flexDirection: 'column',
+    paddingLeft: 20,
+  },
+  heading: { fontSize: 28, fontWeight: 'bold', marginBottom: 20 },
+  permissionRowComment: {},
+  permissionRowContainer: { marginBottom: 15, marginLeft: 30 },
+  permissionRowText: { fontSize: 18 },
+});
+
+const stateToProps = state => {
+  const usingVaccines = selectUsingVaccines(state);
+  const bluetooth = PermissionSelectors.bluetooth(state);
+  const writeStorage = PermissionSelectors.writeStorage(state);
+  const location = PermissionSelectors.location(state);
+  return { bluetooth, location, usingVaccines, writeStorage };
+};
+
+const dispatchToProps = dispatch => {
+  const requestWriteStorage = () => dispatch(PermissionActions.requestWriteStorage());
+  const requestLocation = () => dispatch(PermissionActions.requestLocation());
+  const requestBluetooth = () => dispatch(PermissionActions.requestBluetooth());
+
+  return {
+    requestBluetooth,
+    requestLocation,
+    requestWriteStorage,
+  };
+};
+
+SettingsIconComponent.propTypes = {
+  bluetooth: PropTypes.bool.isRequired,
+  location: PropTypes.bool.isRequired,
+  requestBluetooth: PropTypes.func.isRequired,
+  requestLocation: PropTypes.func.isRequired,
+  requestWriteStorage: PropTypes.func.isRequired,
+  usingVaccines: PropTypes.bool.isRequired,
+  writeStorage: PropTypes.bool.isRequired,
+};
+export const SettingsIcon = connect(stateToProps, dispatchToProps)(SettingsIconComponent);
```

Comment:
I like this component a lot 😄 

Maaaaybe `PermissionSettingsIcon` might be more accurate. Until we put other stuff in there. Also services. 🤷 

Was also thinking of having some sort of extensibility plan - when we need to also have things for barcode scanners n stuff, the amount of conditionals might blow up, - constructing the 'list' of permissions with a builder pattern or something - buuuut, future problems..?

Reply:
Yes. the name `SettingsIcon` is too generic. `PermissionSettingsIcon` works for me, I just worry that having `Services` in it? 

I'm also troubled at having a little cog icon at the top right, and having a settings button with a little cog icon down the bottom right. That's confusing isn't it?

Reply:
And yes - re extensibility. absolutely. My preference is to look at that later 😉  I'll use the excuse of 'this issue is big enough already' although really the reason is 'that sounds like a lot of work and I'm lazy'

...though I would like to try the location services, will have a go at including the other package and testing that now

Reply:
I thought the double cog icons weren't too confusing, the permissions one being next to the sync stuff was enough of an indication it was probably sync related in my mind 🤷 . Could use a wrench to be different if it's weird that there's two similar buttons doing different things? Of course I have to suggest a cat too, but I think that won't help here 😿 .